Mathlib/Data/Nat/Bitwise.lean

```diff
@@ -318,4 +319,58 @@ theorem lt_lxor'_cases {a b c : ℕ} (h : a < lxor' b c) : lxor' a c < b ∨ lxo
   (or_iff_right fun h' => (h.asymm h').elim).1 <| lxor'_trichotomy h.ne
 #align nat.lt_lxor_cases Nat.lt_lxor'_cases
 
+lemma div2_succ_succ (x) :
+    div2 (x + 2) = (div2 x) + 1 := by
+  simp only [div2_succ, bodd_succ, Bool.cond_not]
+  cases bodd x <;> simp only [cond_false, cond_true]
+
+lemma div2_eq_zero {x : ℕ} :
+    div2 x = 0 → x = 0 ∨ x = 1 := by
+  intro h
+  rcases x with ⟨⟩ | ⟨⟩ | x
+  next => apply Or.inl rfl
+  next => apply Or.inr rfl
+  next =>
+    rw[div2_succ_succ] at h
+    contradiction
+
+lemma bitwise'_bit' {f : Bool → Bool → Bool} (a m b n)
```

Comment:
Does `bitwise'_bit` already exist?

Can you write the types of `a m b n` explicitly?

Reply:
`bitwise'_bit` exists, but has `f false false = false` as assumption, we replace that assumption with `(ham : m = 0 → a = true)` and `(hbn : n = 0 → b = true)`.